### ISSUE TITLE
CPT/templates: Inserter horizontal line still shows up

### ISSUE THREAD

**Issue by jasmussen** [WordPress/gutenberg#29211]
## Description

If you create a custom post type with a locked template (`$post_type_object->template_lock`), the sibling inserter is not supposed to show up. But the blue horizontal line still does.

## Step-by-step reproduction instructions

Create a custom post type with a template. Here's something you can drop in your functions.php to test it:

```
// Create a custom post type.
function myplugin_custom_post_type() {
	register_post_type( 'podcast',
		array(
			'labels' => array(
				'name' => __( 'Podcasts' ),
				'singular_name' => __( 'Podcast' )
			),
			'public' => true,
			'has_archive' => true,
			'rewrite' => array('slug' => 'podcast'),
			'show_in_rest' => true,
			'supports' => [
				'title', 'editor', 'custom-fields', // Custom fields must be supported for meta handling to function.
			],
		)
	);
}
add_action( 'init', 'myplugin_custom_post_type' );

// Assign a post template to our custom post type
function myplugin_register_template() {
	$post_type_object = get_post_type_object( 'podcast' );
	$post_type_object->template = array(
		array( 'core/heading', array( 'level' => 3, 'content' => 'Description', ) ),
		array( 'core/paragraph', array( 'placeholder' => 'Add description', ) ),
		array( 'core/heading', array( 'level' => 3, 'content' => 'Heading', ) ),
		array( 'core/paragraph', array( 'placeholder' => 'Add show notes', ) ),
	);

	// Lock the template from being rearranged or items deleted.
	$post_type_object->template_lock = 'all'; // Can be "all" or "insert", the latter allows moving blocks around.
}
add_action( 'init', 'myplugin_register_template' );
```

Be sure that `$post_type_object->template_lock = 'all';` is present, and set to "all" or "insert".

## Expected behaviour

The blue line should not show up.

## Screenshots or screen recording (optional)

![locked](https://user-images.githubusercontent.com/1204802/108699640-c4161580-7505-11eb-93fa-a2e40bbd64ba.gif)

**Comment by Mamaduka** [comment]
I think this got fixed via #32576.

**Comment by jasmussen** [comment]
Confirmed! I'm not seeing any hover lines:

<img width="556" alt="Screenshot 2021-08-04 at 15 26 35" src="https://user-images.githubusercontent.com/1204802/128188979-88b845f0-5e67-4184-83db-b5f1cabfcb5f.png">

Thank you!
